### PR TITLE
feat(SDK): add `globs` and `dir` support for artifact deps.

### PR DESCRIPTION
**Pull request by destifo**
- [x] Include glob and dir support for `PythonRuntime` deps.
- [x] Include glob and dir support for `DenoRuntime` deps.
- [x] add tests

The change includes support for declaring artifact dependencies through `globs` and `dirs`

[MET-441](https://linear.app/metatypedev/issue/MET-441/sdk-support-globs-and-directories-in-artifact-dependencies)

#### Migration notes

In the `deps` parameter for `python.import(...)` and `deno.import(...)`, globs and dirs can be passed in addition to files. 

- [x] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality for defining and deploying Typegraphs for Deno and Python runtimes.
  - Added support for defining a Deno runtime dependency graph with policies for test scenarios.

- **Bug Fixes**
  - Corrected the structure of the `Deno.serve` call in the `serve` function.

- **Refactor**
  - Enhanced method chaining for better readability in the `MetaTest` class.

- **Tests**
  - Updated test coverage reporting to include new Deno runtime test files.
  - Commented out and removed outdated test cases in Deno runtime tests.

- **Chores**
  - Updated platform specification in configuration files.

### CODE REVIEW

File: .ghjk/lock.json [hunk 1]
```diff
@@ -856,25 +855,6 @@
           },
           "packageName": "node-gyp"
         },
-        "88f6aa5ddb513712623d31a17096ea26286064da": {
-          "version": "v2.4.0",
-          "port": {
-            "ty": "denoWorker@v1",
```
Comment (zifeo) [inline .ghjk/lock.json:862]: @Yohe-Am you have a plan or strategy to stabilize the lock file and avoid change when the arch is different?

File: typegraph/core/src/utils/fs_host.rs [hunk 2]
```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
...
+    for dep in deps {
+        if is_glob(&dep) {
+            let _ = get_matching_files(&dep)
+                .unwrap()
+                .into_iter()
+                .map(|file| resolved_deps.push(file));
+        } else {
+            // print(&format!(
+            //     "***************** {:?}",
+            //     make_absolute(&PathBuf::from(dep.clone()))?
```
Comment (zifeo) [inline typegraph/core/src/utils/fs_host.rs:267]: remove the debug comment or can we have log with given level in wasm?

Comment (destifo) [reply typegraph/core/src/utils/fs_host.rs:267]: I will remove the all debug comments in the end all at once. I haven't found any loggers in the core SDK but I will check if that's possible.

**Comment by destifo** [comment]
@CodiumAI-Agent /review

**Review by Yohe-Am — APPROVED** [review]

File: typegraph/core/src/utils/fs_host.rs [hunk 3]
```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
...
+}
+
+pub fn get_dir(path: &str) -> PathBuf {
+    let directory = PathBuf::from(path);
+
+    PathBuf::from(directory.parent().unwrap())
+}
+
+pub fn get_matching_files(glob_pattern: &str) -> Result<Vec<PathBuf>, String> {
+    let mut matching_files = vec![];
```
Comment (afmika) [inline typegraph/core/src/utils/fs_host.rs:239]: Not a prioarity but maybe rename it this to `expand_glob` ? `get_matching_files` is sort of too abstract if I didn't read the arg name to get more context.

File: typegraph/core/src/utils/fs_host.rs [hunk 4]
```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
...
+            matching_files.push(make_relative(&file)?);
+        }
+    }
+
+    // eprint(&format!("***************** {:?}", matching_files));
+    Ok(matching_files)
+}
+
+pub fn resolve_globs_dirs(deps: Vec<String>) -> Result<Vec<PathBuf>, String> {
+    let mut resolved_deps = vec![];
```
Comment (afmika) [inline typegraph/core/src/utils/fs_host.rs:257]: Maybe use a HashSet? Each globs inside `deps` may expand to similar paths.

**Review by afmika — COMMENTED** [review]
Nice, can't wait for this to be merged!

File: typegraph/core/src/utils/fs_host.rs (hunk 4 above)
Comment (destifo) [reply typegraph/core/src/utils/fs_host.rs:257]: good point! they get filtered later on the pipeline but can also do it here.

File: typegate/tests/utils/test.ts [hunk 5]
```diff
@@ -277,6 +277,8 @@ export class MetaTest {
       cwd,
     );
 
+    console.log(serialized);
```
Comment (zifeo) [inline typegate/tests/utils/test.ts:280]: a left over?

File: typegraph/core/src/utils/postprocess/deno_rt.rs [hunk 6]
```diff
@@ -65,7 +70,7 @@ impl DenoProcessor {
 
         let mut tg_deps_paths = vec![];
         let mut tg_artifacts = vec![];
-
+        // eprint(&format!("&&&&&&&&&&&&&&&&&&&&&&&& {:?}", main_path));
```
Comment (zifeo) [inline typegraph/core/src/utils/postprocess/deno_rt.rs:73]: eprint to cleanup?

File: typegraph/core/src/utils/fs_host.rs [hunk 7]
```diff
@@ -214,3 +216,39 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
 pub fn path_exists(path: &Path) -> Result<bool, String> {
     path_exists_host(&path.to_string_lossy())
 }
+
+pub fn is_glob(path: &str) -> bool {
+    let path = PathBuf::from(path);
+    if let Some(base_name) = path.file_name() {
```
Comment (destifo) [inline typegraph/core/src/utils/fs_host.rs:222]: one issue here, only checking the file name for wild cards. will push the fix for this.

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**